### ISSUE TITLE
does distcc support android 10/11 compiling?

### ISSUE THREAD

**Issue by modoojunko** [distcc/distcc#376]
i noticed that distcc can warp C/C++ compile tools to make compile faster.

in android 10 , it switches the compile tool to clang/clang++ and use ninjia to call them via -c:-S:-E and -o options.

i did some tests use distcc to compile failed, but i can not send the log from internel net .

i tried so many days, i want to know does this can work from the begining

**Comment by shawnl** [comment]
It should work with android, but it is not tested. (in non-pump mode) The non-pump mode *only* uses gcc and cpp.

**Comment by modoojunko** [comment]
@shawnl you means distcc can used to compile AOSP with out problem?

**Comment by shawnl** [comment]
No, I thought you meant running distccd on android...

**Comment by modoojunko** [comment]
@shawnl  i means use discc to make compole aosp faster, does this point work?

**Comment by shawnl** [comment]
Distcc server can compile anything that you can compile into `.i` files with gcc, and does not use a few gcc features that require additional files (such as plugins).

**Comment by modoojunko** [comment]
@shawnl AOSP use clang -c;-E-:-S:o options. i tried to use distcc but failed. it only can compile in local server not remote one

**Comment by shawnl** [comment]
Are you using clang's `-t` option correctly?

**Comment by modoojunko** [comment]
@shawnl i did not setup the clang options but AOSP's build tools did . i add distcc to AOSP's allowed compile list . it can work. but i do not get your "whitelist" point . is it a distccd option? i install distcc via apt install distcc. my os is ubuntu 16.04 and distcc version is 3.1

**Comment by shawnl** [comment]
This is a duplicate of Debian bug https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=926884

you can work around it by using the `-target` option.

**Comment by shawnl** [comment]
The documentation of the `whitelist` is in the distcc man page, and was introduced by me in version `3.3`.

**Comment by modoojunko** [comment]
@shawnl i will update my distcc to version 3.3 and tried again

**Comment by shawnl** [comment]
@modoojunko You are not understanding what I am saying, you didn't read the bug I posted, and you seem incapable of following documentation even when the correct documentation is pointed out to you.

**Comment by modoojunko** [comment]
@shawnl of cause i read it , i know it is a bug in debian source. and it till not fixed? i do not have enough knowledge to understand the content of the email. if i  install distcc 3.3.3 via source code instead of apt  , that  should be ok?

**Comment by modoojunko** [comment]
@shawnl  today, i install distcc 3.3.2 version via source code. run distccd with --enable-tcp-insecure. the result is bad. the remote compile on distccd always failed and retry it on local success.

**Comment by modoojunko** [comment]
@shawnl In these days , I tried so many times. As you mentioned the "-target" option of clang , I set DISTCC_VERBOSE=1 and use distcc in AOSP compilation via "distcc clang++ -i -o...." way. I noticed that , in the compilation the clang option "--target" is "aarch64-linux-android", it is not the value of GNU_HOST when install distcc which is "x86_64-pc-linux-gnu". I want to know that ,if i modify GNU_HOST in the src/config.h as same as "aarch64-linux-android" this can work?
